aws-frauddetector-detector/src/aws_frauddetector_detector/handler_workers.py

```diff
@@ -78,7 +78,32 @@ def execute_update_detector_handler_work(session: SessionProxy,
 
 
 def execute_delete_detector_handler_work(session: SessionProxy, model: models.ResourceModel, progress: ProgressEvent):
-    pass
+    afd_client = client_helpers.get_singleton_afd_client(session)
+
+    # For contract_delete_delete, we need to fail if the resource DNE
+    get_detectors_works, _ = validation_helpers.check_if_get_detectors_succeeds(afd_client, model.DetectorId)
+    if not get_detectors_works:
+        raise exceptions.NotFound('detector', model.DetectorId)
+
+    try:
```

Comment:
we definitely will want testing on code blocks like this

Reply:
Agreed. I have been using contract tests as a source of confidence in the interim, and the generated rpdk logs.

I can add unit tests into this PR if you'd like the extra confidence, otherwise I'd like to get all the handlers in, then update all unit tests in one go.

Reply:
Ill approve but we should try to get testing in as close to the feature development as possible so that there arent a bunch of surprises when we build the tests

Reply:
Makes sense. If it weren't for the contract tests and the logs, I would not defer unit tests like this. I'll include them in the next PR.